src/font/Collection.zig

```diff
@@ -93,6 +93,13 @@ pub fn add(
 ) AddError!Index {
     const list = self.faces.getPtr(style);
 
+    // If we error here, we're responsible for
+    // deiniting the face that we were passed.
+    errdefer {
+        var f = face;
+        f.deinit();
+    }
```

Comment:
This block is suspicious. We do take ownership of the face but usually in APIs like this I've found this only applies _when it succeeds_. If it doesn't succeed I think it's fair to let the caller determine what they want to do with the face (it isn't clear to me that they definitely want it reinitialized).

Reply:
Yeah I added this because of a failing test (`Collection.test.add full`), but it could just be changed to handle it at any call site if you think that'd be better

Reply:
I think that would be more functionally correct, though perhaps noisier or more annoying to implement in tests. unsure. But I think we should do it. I'm happy to do that cleanup myself if you don't get to it.